src/frontend/Module.php

```diff
@@ -115,11 +118,25 @@ public function registerComponents()
     public function luyaBootstrap(Application $app)
     {
         if (!$app->request->isConsoleRequest && !$app->request->isAdmin) {
+    
+            if ($app->has('composition') && $app->has('website')) {
+                if (empty($app->composition->hostInfoMapping)) {
+                    $app->composition->hostInfoMapping = $app->website->createHostInfoMapping();
+                }
+            }
+            
             if ($app->has('themeManager')) {
                 // set active theme from database
                 $app->get('themeManager')->on(ThemeManager::EVENT_BEFORE_SETUP, function (SetupEvent $event) {
-                    // get the base path of the active theme
-                    $activeBasePath = Theme::find()->cache()->select('base_path')->where(['is_active' => 1])->scalar();
+                    /**
+                     * get the base path of the website theme
+                     * @since 4.0.0
+                     */
+                    $activeBasePath = Theme::find()->cache()->select('base_path')->where(['id' => \Yii::$app->website->current['theme_id']])->scalar();
```

Comment:
Does `\Yii::$app->website->current['theme_id']` resolve always in a value? this looks very error prone at first sight.

Reply:
Why does ->current do not resolve an object? current->theme_id?

Reply:
I added a if condition. `current` is the array result from `\luya\cms\Website::loadAllWebsiteData`